app/controllers/shopping_controller.rb

```diff
@@ -0,0 +1,14 @@
+class ShoppingController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @recipes = current_user.recipes
+
+    @foods = Food.where(user_id: current_user.id)
+
+    @total_price = 0
+    @recipes.each do |recipe|
+      @total_price += recipe.total_price
```

Comment:
**Optional Change**
- [ ] It is good to keep the controller as lean as possible. This line could be a method/scope in the Shopping model and just called in the controller.